apps/meteor/ee/app/livechat-enterprise/server/hooks/beforeNewInquiry.ts

```diff
@@ -0,0 +1,67 @@
+import { Meteor } from 'meteor/meteor';
+import { LivechatPriority, OmnichannelServiceLevelAgreements } from '@rocket.chat/models';
+import type { ILivechatInquiryRecord, ILivechatPriority, IOmnichannelServiceLevelAgreements } from '@rocket.chat/core-typings';
+
+import { callbacks } from '../../../../../lib/callbacks';
+import { cbLogger } from '../lib/logger';
+
+type Props = {
+	sla?: string;
+	priority?: string;
+	[other: string]: any;
+};
+
+const beforeNewInquiry = async (extraData: Props) => {
+	const { sla: slaSearchTerm, priority: prioritySearchTerm, ...props } = extraData;
+	if (!slaSearchTerm && !prioritySearchTerm) {
+		cbLogger.debug('Skipping callback. No sla or priority provided');
+		return extraData;
+	}
+
+	let sla: IOmnichannelServiceLevelAgreements | null = null;
+	let priority: ILivechatPriority | null = null;
+
+	if (slaSearchTerm) {
+		sla = await OmnichannelServiceLevelAgreements.findOneByIdOrName(slaSearchTerm, {
+			projection: { dueTimeInMinutes: 1 },
+		});
+		if (!sla) {
+			throw new Meteor.Error('error-invalid-sla', 'Invalid sla', {
+				function: 'livechat.beforeInquiry',
+			});
+		}
+	}
+	if (prioritySearchTerm) {
+		priority = await LivechatPriority.findOneByIdOrName(prioritySearchTerm, {
+			projection: { _id: 1, sortItem: 1 },
+		});
+		if (!priority) {
+			throw new Meteor.Error('error-invalid-priority', 'Invalid priority', {
+				function: 'livechat.beforeInquiry',
+			});
+		}
+	}
+
+	const now = new Date();
+	const ts = new Date(now.getTime());
```

Comment:
any special reason for doing this? I guess this is exactly the same, isn't it?
```suggestion
	const ts = new Date();
```

Reply:
They had another use before the changes, so they're leftovers